source/extensions/common/tap/admin.cc

```diff
@@ -85,10 +85,10 @@ Http::Code AdminHandler::handler(absl::string_view, Http::HeaderMap&, Buffer::In
   return Http::Code::OK;
 }
 
-Http::Code AdminHandler::badRequest(Buffer::Instance& response, absl::string_view error) {
-  ENVOY_LOG(debug, "handler bad request: {}", error);
-  response.add(error);
-  return Http::Code::BadRequest;
+Http::Code AdminHandler::badRequest(Buffer::Chunker& response, absl::string_view error) {
+  absl::string_view errorString = absl::string_view{"handler bad request: "} + error;
+  ENVOY_LOG(debug, errorString);
+  response.reportError(Http::Code::BadRequest, errorString) return Http::Code::BadRequest;
```

Comment:
one thing to be careful of is if we are going to be responding with HTML we need to sanitize untrusted input. E.g. if an attacker can control the contents of 'error' somehow they could put an XSS in there.

Reply:
Is there a existed api to do it(sorry that I am not familiar with that)?